src/gtk/caret.cpp

```diff
@@ -0,0 +1,205 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        src/gtk/caret.cpp
+// Purpose:     GTK3 implementation of wxCaret
+// Author:      Kettab Ali
+// Created:     2022-01-29
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// For compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#if wxUSE_CARET && defined(__WXGTK3__)
+
+#ifndef WX_PRECOMP
+    #include "wx/window.h"
+#endif // WX_PRECOMP
+
+#include "wx/caret.h"
+#include "wx/gtk/private/wrapgtk.h"
+
+// ===========================================================================
+// implementation
+// ===========================================================================
+
+namespace wxPrivate
+{
+static bool IsWayland()
```

Comment:
This should really go to some private function instead of being checked in a half a dozen of different places.

Reply:
I think this should be done later in a separate PR instead of here.